superset/assets/src/dashboard/components/filterscope/FilterScopeSelector.jsx

```diff
@@ -0,0 +1,518 @@
+    return (
+      <FilterFieldTree
+        activeKey={activeKey}
+        nodes={filterFieldNodes}
+        checked={checkedFilterFields}
+        expanded={expandedFilterIds}
+        onClick={this.onChangeFilterField}
+        onCheck={this.onCheckFilterField}
+        onExpand={this.onExpandFilterField}
+      />
+    );
+  }
+
+  renderFilterScopeTree() {
+    const {
+      filterScopeMap,
+      activeKey,
+      isSingleEditMode,
+      searchText,
+    } = this.state;
+
+    const selectedFilterId = isSingleEditMode
+      ? getChartIdAndColumnFromFilterKey(activeKey).chartId
+      : 0;
+    return (
+      <React.Fragment>
+        <input
+          className={cx('filter-text scope-search', {
+            'multi-edit-mode': !isSingleEditMode,
+          })}
+          placeholder={t('Search...')}
+          type="text"
+          value={searchText}
+          onChange={this.onSearchInputChange}
+        />
+        <FilterScopeTree
+          nodes={filterScopeMap[activeKey].nodesFiltered}
+          checked={filterScopeMap[activeKey].checked}
+          expanded={filterScopeMap[activeKey].expanded}
+          onCheck={this.onCheckFilterScope}
+          onExpand={this.onExpandFilterScope}
+          // pass selectedFilterId prop to FilterScopeTree component,
+          // to hide checkbox for selected filter field itself
+          selectedFilterId={selectedFilterId}
+        />
+      </React.Fragment>
+    );
+  }
+
+  renderEditModeControl() {
+    const { isSingleEditMode } = this.state;
+    return (
+      <span
+        role="button"
+        tabIndex="0"
+        className="edit-mode-toggle"
+        onClick={this.onToggleEditMode}
+      >
+        {isSingleEditMode
+          ? t('Edit multiple filters')
+          : t('Edit individual filter')}
+      </span>
+    );
+  }
+
+  render() {
+    const { dashboardFilters } = this.props;
+    const { showSelector, isSingleEditMode, activeKey } = this.state;
+    const isSingleValue = activeKey.indexOf('[') === -1;
+    const currentFilterLabels = []
+      .concat(isSingleValue ? activeKey : JSON.parse(activeKey))
+      .map(key => {
+        const { chartId, column } = getChartIdAndColumnFromFilterKey(key);
+        return dashboardFilters[chartId].labels[column] || column;
+      });
+
+    return (
+      <React.Fragment>
+        <div className="filter-scope-container">
+          <div className="filter-scope-header">
+            <h4>{t('Configure filter scopes')}</h4>
+            <div
+              className={cx('selected-fields', {
+                'multi-edit-mode': !isSingleEditMode,
+              })}
+            >
+              {`Batch editing ${currentFilterLabels.length} filters: `}
+              <span className="selected-scopes">
+                {currentFilterLabels.join(', ')}
+              </span>
+            </div>
+          </div>
+
+          {!showSelector ? (
+            <div>{t('There are no filter in this dashboard')}</div>
```

Comment:
grammar nit: `There are no filters in this dashboard`

Reply:
fixed.